src/app/components/room/room.service.js

```diff
@@ -106,10 +106,33 @@
       var $$rootScope = $rootScope;
       var connection = null;
 
+      // adding username to jhEventsProvider
+      jhEventsProvider.username = username;
+      // adding room details to jhEventsProvider
+      jhEventsProvider.roomDesc = that.room.description;
```

Comment:
I wouldn't initialize this here. I would make the callstats provider ask directly the information it needs to `UserService` and `RoomService`. This is another unnecessary coupling of our event emitter with the callstats take on things.

Reply:
@code-master5 you told me about this being not possible. But I did some experiments with the logic of `jsEventsProvider` and `Callstats` being placed in new components within the `janusHangouts` module (instead of defining their own modules) and I don't see why shouldn't it work.

Reply:
@ancorgs Sorry, I can't understand. Can you provide me the link to your experiments for reference? Things were not working in my case! May be I missed something here.